packages/easy-coding-standard/src/Console/Output/JsonOutputFormatter.php

```diff
@@ -41,7 +41,9 @@ public function report(ErrorAndDiffResult $errorAndDiffResult, int $processedFil
         $this->easyCodingStandardStyle->writeln($json);
 
         $errorCount = $errorAndDiffResult->getErrorCount();
-        return $errorCount === 0 ? ShellCode::SUCCESS : ShellCode::ERROR;
+        $diffCount = $errorAndDiffResult->getFileDiffsCount();
+
+        return $errorCount + $diffCount === 0 ? ShellCode::SUCCESS : ShellCode::ERROR;
```

Comment:
This is actually incosistent with `ConsoleOutputFormatter` now. It returns success on diffs.

The design should not be handed here but via `--dry-run`, as I described before.